crates/bevy_ecs/src/system/query.rs

```diff
@@ -640,6 +682,208 @@ where
+    /// assert_eq!(*a_iterator.next().unwrap(), A(1));
+    /// ```
+    #[inline]
+    pub fn get_multiple(
+        &'s self,
+        entities: impl IntoIterator<Item = Entity>,
+    ) -> impl Iterator<Item = Result<<Q::ReadOnlyFetch as Fetch>::Item, QueryEntityError>> {
+        entities.into_iter().map(|entity| self.get(entity))
+    }
+
+    /// Returns the query results for the ['BTreeSet'](std::collections::BTreeSet) of [`Entity`]s provided.
+    ///
+    /// These values follow the order of your input iterator (if any).
+    /// In case of nonunique entities, a nonexisting entity or a mismatched component,
+    /// a [`QueryEntityError`] is returned instead.
+    ///
+    /// If you need to verify the identity of each item returned,
+    /// add [`Entity`] to your [`Query`].
+    ///
+    /// If you absolutely cannot afford the overhead of verifying uniqueness in this way,
+    /// you can (carefully) call the unsafe [`get_unchecked`](Self::get_unchecked) method repeatedly instead.
+    ///
+    /// If you need to get the data from exactly two specific entities at once,
+    /// use ['get_pair_mut'](Self::get_pair_mut).
+    ///
+    /// # Example
+    /// ```rust
+    /// # use bevy_ecs::prelude::*;
+    /// #[derive(Component, PartialEq, Debug)]
+    /// struct A(u64);
+    ///
+    /// let mut world = World::new();
+    /// let entity_1 = world.spawn().insert(A(1)).id();
+    /// let entity_2 = world.spawn().insert(A(2)).id();
+    /// let entity_3 = world.spawn().insert(A(3)).id();
+    ///
+    /// let query_state = world.query::<&mut A>();
+    /// let mut a_query = Query::from_state(&mut world, &query_state);
+    /// let mut a_iterator = a_query.get_multiple_mut([entity_1, entity_3]).map(|i|i.unwrap());
+    /// let mut a_1 = a_iterator.next().unwrap();
+    /// let mut a_3 = a_iterator.next().unwrap();
+    ///
+    /// *a_1 = A(11);
+    /// *a_3 = A(33);
+    ///
+    /// // Manually drop references so we can access the `World` again
+    /// std::mem::drop(a_iterator);
+    /// std::mem::drop(a_query);
+    ///
+    /// assert_eq!(*world.get::<A>(entity_1).unwrap(), A(11));
+    /// assert_eq!(*world.get::<A>(entity_2).unwrap(), A(2));
+    /// assert_eq!(*world.get::<A>(entity_3).unwrap(), A(33));
+    /// ```
+    #[inline]
+    pub fn get_multiple_mut(
+        &mut self,
+        entities: impl IntoIterator<Item = Entity>,
+    ) -> impl Iterator<Item = Result<<Q::Fetch as Fetch>::Item, QueryEntityError>> {
+        let entities_iter = entities.into_iter();
+
+        // Preallocating the HashSet used to check uniqueness based on the expected maximum amount of space
+        let (lower_bound, maybe_upper_bound) = entities_iter.size_hint();
+        let best_bound = if let Some(upper_bound) = maybe_upper_bound {
+            upper_bound
+        } else {
+            lower_bound
+        };
+
+        let entities_seen = HashSet::with_capacity(best_bound * std::mem::size_of::<Entity>());
```

Comment:
If I am not misunderstanding the intent of this call (please correct me if I am, still new here!), I think we don’t need the size_of call as the method should be doing it internally at least that’s what I understood from [the documentation for with_capacity](https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.with_capacity) [emphasis mine]
> The hash set will be able to hold at least `capacity` **elements** without reallocating.

```suggestion
        let entities_seen = HashSet::with_capacity(best_bound);
```

Reply:
Ah, so you're right: it automatically looks at the size of its storage: I'd assumed that it was taking a size in bytes. Thanks!